### PR TITLE
Fix --ignore-running not working with -p/-t in delete cmd

### LINKED ISSUE #1571: Running PRs and TRs getting deleted with --ignore-running=true
# Versions and Operating System

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  Kubernetes 1.23.5  
  OpenShift 4.10
  ```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

  ```
  Client version: 0.23.1
  Pipeline version: v0.33.2
  ```

- Operating System:

  ```
  Fedora 35
  ```

# Expected Behavior

When using `--ignore-running=true` with `-p=<pipeline-name` in `tkn pr delete` or `tkn tr delete`, it should not delete the running pipelineruns or taskruns.

# Actual Behavior

It is deleting all the pipelineruns or taskruns of that respective pipeline or task, without taking a note of the running state.

# Steps to Reproduce the Problem

1. Create a few PR of a pipeline with some `sleep 10000`
2. Hit `tkn pr delete --p=test --ignore-running=true`
3. Hit `tkn pr ls` and you will see all the running prs got deleted.

# Additional Info

It is working fine with `--all` flag but not with `-p=<pipeline-name` flag

### PR DESCRIPTION
**Pull request by piyush-garg**
This will add the fix to --ignore-running=true is not working with
-p/-t/--clustertask in delete cmd resulting in deleting all runs.

A bit of refactoring and adds unit tests

Fix #1571

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Fix --ignore-running not working with -p/-t in delete cmd
```

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: pkg/cmd/taskrun/delete_test.go [hunk 1]
```diff
@@ -261,6 +261,120 @@ func TestTaskRunDelete(t *testing.T) {
...
+				Name:      "tr0-8",
+				Namespace: "ns",
+				Labels:    map[string]string{"tekton.dev/task": "random"},
+			},
+			Spec: v1alpha1.TaskRunSpec{
+				TaskRef: &v1alpha1.TaskRef{
+					Name: "random",
+					Kind: v1alpha1.NamespacedTaskKind,
+				},
+			},
```
[the remark this answers is not in the document]
Comment (piyush-garg) [reply pkg/cmd/taskrun/delete_test.go:275]: for now i added them, but if we want we can create an epic and remove in the codebase

File: pkg/cmd/taskrun/delete_test.go [hunk 2]
```diff
@@ -569,6 +687,60 @@ func TestTaskRunDelete(t *testing.T) {
...
+			want:        "All TaskRuns(Completed) associated with Task \"random\" deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "Delete all of task with explicit --ignore-running true",
+			command:     []string{"delete", "-t", "random", "-f", "-n", "ns", "--ignore-running=true"},
+			dynamic:     seeds[10].dynamicClient,
+			input:       seeds[10].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All TaskRuns(Completed) associated with Task \"random\" deleted in namespace \"ns\"\n",
```
Comment (vinamra28) [inline pkg/cmd/taskrun/delete_test.go:706]: seems to be duplicate as the default value of `--ignore-running` is true.

File: pkg/cmd/taskrun/delete_test.go [hunk 3]
```diff
@@ -569,6 +687,60 @@ func TestTaskRunDelete(t *testing.T) {
...
+			inputStream: nil,
+			wantError:   false,
+			want:        "All TaskRuns(Completed) associated with ClusterTask \"random\" deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "Delete all of clustertask with explicit --ignore-running true",
+			command:     []string{"delete", "--clustertask", "random", "-f", "-n", "ns", "--ignore-running=true"},
+			dynamic:     seeds[13].dynamicClient,
+			input:       seeds[13].pipelineClient,
+			inputStream: nil,
```
Comment (vinamra28) [inline pkg/cmd/taskrun/delete_test.go:731]: same

File: pkg/cmd/taskrun/delete_test.go [hunk 4]
```diff
@@ -1220,6 +1510,60 @@ func TestTaskRunDelete_v1beta1(t *testing.T) {
...
+			input:       seeds[11].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All TaskRuns(Completed) associated with Task \"random\" deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "Delete all of task with explicit --ignore-running true",
+			command:     []string{"delete", "-t", "random", "-f", "-n", "ns", "--ignore-running=true"},
+			dynamic:     seeds[12].dynamicClient,
+			input:       seeds[12].pipelineClient,
```
Comment (vinamra28) [inline pkg/cmd/taskrun/delete_test.go:1526]: same

File: pkg/cmd/taskrun/delete_test.go [hunk 5]
```diff
@@ -1220,6 +1510,60 @@ func TestTaskRunDelete_v1beta1(t *testing.T) {
...
+			command:     []string{"delete", "--clustertask", "random", "-f", "-n", "ns", "--ignore-running"},
+			dynamic:     seeds[14].dynamicClient,
+			input:       seeds[14].pipelineClient,
+			inputStream: nil,
+			wantError:   false,
+			want:        "All TaskRuns(Completed) associated with ClusterTask \"random\" deleted in namespace \"ns\"\n",
+		},
+		{
+			name:        "Delete all of clustertask with explicit --ignore-running true",
+			command:     []string{"delete", "--clustertask", "random", "-f", "-n", "ns", "--ignore-running=true"},
```
Comment (vinamra28) [inline pkg/cmd/taskrun/delete_test.go:1551]: same

File: pkg/cmd/taskrun/delete_test.go (hunk 2 above)
Comment (piyush-garg) [reply pkg/cmd/taskrun/delete_test.go:706]: Added it to check the default behavior, if somehow the default behavior changes in future

**Review by pradeepitm12 — APPROVED** [review]
/lgtm

**Pull request merged**